intern/rest.py

```diff
@@ -0,0 +1,85 @@
+from rest_framework import filters
+from rest_framework import viewsets
+from rest_framework import status
+from rest_framework.permissions import DjangoModelPermissions
+from rest_framework.response import Response
+
+from intern.models import *
+from intern.serializers import InternRoleFullSerializer, InternSerializer, AccessLevelSerializer, InternGroupSerializer, \
+    RoleSerializer, AddInternRoleSerializer
+
+
+class InternViewSet(viewsets.ModelViewSet):
+    permission_classes = (DjangoModelPermissions,)
+    search_fields = ('user__username', 'user__realname')
+    filter_backends = (filters.SearchFilter, filters.DjangoFilterBackend)
+    filter_fields = ('user', 'user__username')
+
+    def get_queryset(self):
```

Comment:
You can simplify all this by removing the functions and settings `queryset` on the class. Same for `serializer_class`. The function would be needed if there is any logic needed, but there is none here.